swap/src/bin/swap.rs

```diff
@@ -67,6 +67,12 @@ async fn main() -> Result<()> {
             let swap_id = Uuid::new_v4();
 
             cli::tracing::init(debug, json, data_dir.join("logs"), Some(swap_id))?;
+
+            match check_latest_version().await {
```

Comment:
Couldn't we match and print this error in the `check_latest_version` function? We're not using the err for anything else and that would avoid some of this duplication.

Reply:
I'll make a note to change this and I will see how it looks. Thank you!